scripts/caravan/caravan-gui.lua

```diff
@@ -350,6 +568,154 @@ local function get_caravan_gui_in_flow(player)
 end
 
 function Caravan.get_caravan_gui(player)
-    local gui = get_caravan_gui_in_flow(player) or player.gui.screen.caravan_gui
+    local gui = player.gui.relative.caravan_gui or player.gui.screen.caravan_gui
     if gui then return gui end
 end
+
+function Caravan.get_interrupt_gui(player)
+    -- if player.gui.relative.caravan_flow then
+        return player.gui.relative.py_edit_interrupt_gui
+    -- end
+end
+
+---Creates a GUI for editing the interrupt
+function Caravan.build_interrupt_gui(player, interrupt_name)
```

Comment:
I don't like parameters on this method. It should either include the caravan's `LuaEntity` or the `caravan_data`.
Having it based only on the interrupt_name causes many downstream issues where all called functions inside this function must now support `nil` for the caravan_data.

Reply:
Resolved.